src/app/useRouteRedirects.tsx

```diff
@@ -9,7 +9,17 @@ export const useRouteRedirects = () => {
   const navigate = useNavigate()
 
   useEffect(() => {
-    if (address) {
+    const hasRecentlyGrantedUsbAccess =
```

Comment:
would try `chrome.runtime.sendMessage`  and `chrome.runtime.onMessage.addListener` instead of this logic in hook

Reply:
For that, mv3 flow without service worker would have to be:
- keep ConnectDevice popup open
- click wallet to open default popup
- then default popup can listen to messages